src/frontend/src/optimizer/plan_node/plan_base.rs

```diff
@@ -62,6 +62,9 @@ pub struct StreamExtra {
     /// The watermark column indices of the PlanNode's output. There could be watermark output from
     /// this stream operator.
     watermark_columns: FixedBitSet,
+
+    /// The monotonic columns of the PlanNode's output, Vector of (column index, Monotonicity)
+    monotonic_columns: Vec<(usize, Monotonicity)>,
```

Comment:
What about using new-type pattern (e.g. `struct MonotonicColumn(usize, Monotonicity)`) instead of using anonymous tuple?